app/src/main/java/io/github/gsantner/memetastic/activity/SettingsActivity.java

```diff
@@ -140,8 +146,48 @@ public boolean onPreferenceTreeClick(PreferenceScreen screen, Preference prefere
                         getActivity().finish();
                     }
                 }
+                if (key.equals(getString(R.string.pref_key__is_show_in_gallery))) {
+                    boolean showInGallery = settings.getDefaultPreferences().getBoolean(key, true);
+                    File memeDirectory = AssetUpdater.getMemesDir(AppSettings.get());
+                    File noMediaFile = new File(memeDirectory, ".nomedia");
+                    if (showInGallery) {
+                        noMediaFile.delete();
+                        context.sendBroadcast(new Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE, Uri.fromFile(new File(memeDirectory, ".nomedia"))));
+                        File[] files = memeDirectory.listFiles();
+                        for (int i = 0; i < files.length; i++) {
```

Comment:
Maybe move this to a method addfilestomediastore(). And then both add/deletemediastore methods to an util class. Other than that I think it's quite nice. Does it work on your device?

Reply:
Yeah, I have tested it and it works. I've also tested it on an emulator.